src/overlays/actors/ovl_Obj_Yasi/z_obj_yasi.c

```diff
@@ -28,25 +28,69 @@ const ActorInit Obj_Yasi_InitVars = {
     (ActorFunc)ObjYasi_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80BB4D90[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 800, ICHAIN_STOP),
 };
 
-#endif
+void ObjYasi_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjYasi* this = THIS;
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 0);
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_obj_yasi_Colheader_001428);
 
-extern InitChainEntry D_80BB4D90[];
+    this->dyna.actor.home.rot.y = 0;
 
-extern UNK_TYPE D_06000360;
-extern UNK_TYPE D_06001428;
+    if ((this->dyna.actor.params & 1) != 0) {
+        this->dyna.actor.scale.x = 0.2f;
+        this->dyna.actor.scale.z = 0.2f;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Yasi/ObjYasi_Init.s")
+void ObjYasi_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjYasi* this = THIS;
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Yasi/ObjYasi_Destroy.s")
+void ObjYasi_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjYasi* this = (ObjYasi*)thisx;
+    s16 temp;
+    Vec3f sp1C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Yasi/ObjYasi_Update.s")
+    if (this->dyna.actor.home.rot.z != 0) {
+        if (this->dyna.actor.params < 0) {
+            if (Rand_ZeroOne() < 0.5f) {
+                sp1C.x = this->dyna.actor.world.pos.x;
+                sp1C.y = this->dyna.actor.world.pos.y + 280.0f;
+                sp1C.z = this->dyna.actor.world.pos.z;
+                Item_DropCollectible(globalCtx, &sp1C, ITEM00_NUTS_1);
+            }
+        }
+        this->dyna.actor.home.rot.y = GET_PLAYER(globalCtx)->actor.shape.rot.y;
+        this->dyna.actor.home.rot.x = 400;
+        this->dyna.actor.home.rot.z = 0;
+    }
+    temp = (s16)(this->dyna.actor.shape.rot.x * 0.1f);
+    this->dyna.actor.home.rot.x -= temp;
+    this->dyna.actor.shape.rot.x = this->dyna.actor.shape.rot.x +
+                                   ((s16)(this->dyna.actor.home.rot.x - ((s16)(this->dyna.actor.shape.rot.x * 0.08f))));
```

Comment:
Would using `+=` here match?
Also, the other operand looks like the `BINANG_SUB` macro
```suggestion
    this->dyna.actor.shape.rot.x += BINANG_SUB(this->dyna.actor.home.rot.x, (s16)(this->dyna.actor.shape.rot.x * 0.08f));
```

Reply:
You forgot to add the macro to the decompme 
https://decomp.me/scratch/644QW

Reply:
of course. Thanks :) looks good now!